### PR TITLE
Fixes #21644: add basic react RH repositories page.

### PR DESCRIPTION
**Pull request by waldenraines**
First pass at initial react based RH repositories page.

http://projects.theforeman.org/issues/21644

### CODE REVIEW

**Comment by akofink** [comment]
How do I enable this for testing? I ran `bundle update`, `npm install`, `webpack:compile`, but it's still the old page.

**Comment by danseethaler** [comment]
@akofink if you go to the Lab Features tab then Content Experimental UI then click on the kebab on the right to select RH Repos, that should do it.

**Comment by akofink** [comment]
Thanks @danseethaler. Where is the Lab Features tab?

**Comment by danseethaler** [comment]
@akofink it has to be enabled first by going to the Administer Tab > Settings > Show Experimental Labs -> true :)
You may need to reboot rails afterwards but then the experimental tab will show in the main tabs on the left.

**Comment by danseethaler** [comment]
@waldenraines I think we need to install some dependencies that are included in Foreman but aren't available when we're running tests.

`npm i redux react-redux`
`npm i -D babel-plugin-transform-class-properties`

We'll also need to add `transform-class-properties` as a plugin in `.babelrc`

**Comment by akofink** [comment]
@waldenraines Is this WIP? It's working for me, but it's a blank shell without content.

**Comment by waldenraines** [comment]
> @waldenraines Is this WIP? It's working for me, but it's a blank shell without content.

Yes, it's very much a WIP.  This is just the "shell".

**Comment by akofink** [comment]
Ok, ping me when you want me to re-test.

**Comment by waldenraines** [comment]
> Ok, ping me when you want me to re-test.

What I meant was that this isn't ever going to be the entire page, it's just the "outline" of the page.  We have several more things to do before the page is fully operational.  See http://projects.theforeman.org/issues/21151

**Comment by akofink** [comment]
Oh :rofl:

**Comment by ehelms** [comment]
Given this is the first exposure for a lot of developers to React code, it might have been nice to have this open for more than 24 hours to give some chance to review, ask questions and digest what this code looks like.

**Comment by danseethaler** [comment]
@ehelms I think the rush was that @waldenraines is on PTO for a week and wanted to get this first layout in so it could be refined and used while he's out. I'll be working on this page over the next week so feel free to share any thoughts here or in other upcoming PRs. There are a bunch for this page ([see here](http://projects.theforeman.org/issues/21151)).

File: webpack/components/MultiSelect/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { FormGroup, ControlLabel } from 'react-bootstrap';
```
Comment (ehelms) [inline webpack/components/MultiSelect/index.js]: What does the empty line between sets of imports imply?

File: webpack/components/RedHatRepository/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+RedHatRepository.propTypes = {
+  redHatRepository: PropTypes.object
+};
+
+RedHatRepository.defaultProps = {
+  redHatRepository: {}
+};
+
+
```
Comment (ehelms) [inline webpack/components/RedHatRepository/index.js]: Extra line here?

File: webpack/components/RedHatRepositorySet/index.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import { ListViewExpandableItem, ListViewItem } from 'patternfly-react';
+
+const RedHatRepositorySet = ({ redHatRepositorySet }) => {
+  const renderRepositoryList = () => (
+    redHatRepositorySet.repositories.map(redHatRepository =>
+      <ListViewItem key={redHatRepository.id} heading={redHatRepository.arch} />
+    )
+  );
+
+  // TODO: combine this and the one in RedHatRepository
```
Comment (ehelms) [inline webpack/components/RedHatRepositorySet/index.js]: Is there something that will extract these TODOs? Or should these be Redmine issues to actually track them?

File: webpack/index.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
+import { applyMiddleware } from 'redux';
+
 import componentRegistry from 'foremanReact/components/componentRegistry';
 import { mount } from 'foremanReact/common/MountingService';
 import ExperimentalUi from './containers/Application/index';
 
+
```
Comment (ehelms) [inline webpack/index.js]: Intentional additional empty line?

File: webpack/scenes/Repos/index.js [hunk 5]
```diff
@@ -1,3 +1,134 @@
...
+  static propTypes = {
+    // loadRedHatRepositories: PropTypes.func.isRequired,
+    // loadRedHatRepositorySets: PropTypes.func.isRequired,
+    redHatRepositoriesResponse: PropTypes.object,
+    redHatRepositorySetsResponse: PropTypes.object
+  };
+
+  // componentWillMount() {
+  //   loadData(this.props);
+  // }
```
Comment (ehelms) [inline webpack/scenes/Repos/index.js]: Should this just be removed?

File: webpack/scenes/Repos/index.js [hunk 6]
```diff
@@ -1,3 +1,134 @@
...
+//     redHatRepositoriesResponse: state.redHatRepositories,
+//     redHatRepositorySetsResponse: state.redHatRepositorySets
+//   };
+//   return props;
+// };
+//
+// export default connect(mapStateToProps, {
+//   loadRedHatRepositories,
+//   loadRedHatRepositorySets
+// })(RedHatRepositoriesPage);
```
Comment (ehelms) [inline webpack/scenes/Repos/index.js]: Should this just be removed?

**Comment by ehelms** [comment]
Looks like `webpack/components/SearchInput/tests/index.tests.js` is just an empty file committed? Intentional?

**Comment by danseethaler** [comment]
@ehelms yeah there is a lot of cleanup here. It probably would have been wise to have @waldenraines just push to a branch and keep working on that until a basic page was more fleshed out. I'll get things cleaned up in another PR.

**Pull request merged**